src/oauth/serializers.py

```diff
@@ -212,45 +196,75 @@ def validate(self, attrs, retry=0):
             login.save(request, connect=True)
 
         login_user = login.account.user
-        attrs['user'] = login_user
-        tracked_login = events_api.track_login(login_user, '$success', request)
+        attrs["user"] = login_user
+        tracked_login = events_api.track_login(login_user, "$success", request)
         update_user_risk_score(login_user, tracked_login)
 
         try:
-            visits = WebsiteVisits.objects.get(uuid=attrs['uuid'])
-            visits.user = attrs['user']
+            visits = WebsiteVisits.objects.get(uuid=attrs["uuid"])
+            visits.user = attrs["user"]
             visits.save()
         except Exception as e:
             print(e)
             pass
 
         try:
-            referral_code = attrs.get('referral_code')
+            referral_code = attrs.get("referral_code")
             if referral_code:
                 referral_user = User.objects.get(referral_code=referral_code)
-                user = attrs['user']
-                if referral_code and not user.invited_by and referral_user.id != user.id:
+                user = attrs["user"]
+
+                invited_by_flag_not_set = (
+                    referral_code
+                    and not user.invited_by
+                    and referral_user.id != user.id
+                )
+                if invited_by_flag_not_set:
```

Comment:
This part is new.
This seems to be the best place to pay the invited user bonus.

- Tried using the various allauth signals - request object does not include `referral_code`
- Tried using user `post_save` - but problem is that `invited_by` is set after initial create

Next week, the logic will be moved elsewhere gated by profile claim.